### PR TITLE
ROC-2486 Legal Rep Details stored in cookie

### PR DESCRIPTION
**Pull request by robertparkinson**
https://tools.hmcts.net/jira/browse/ROC-2486

### CODE REVIEW

File: src/main/app/forms/models/representativeDetails.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { FeeAccount } from 'forms/models/feeAccount'
+import * as express from 'express'
+import CookieProperties from 'common/cookieProperties'
+
+export class RepresentativeDetails {
+  organisationName: OrganisationName = new OrganisationName()
+  address: Address = new Address()
+  contactDetails: ContactDetails = new ContactDetails()
+  feeAccount: FeeAccount = new FeeAccount()
+  cookieName: string = ''
```
Comment (timja-kainos) [inline src/main/app/forms/models/representativeDetails.ts]: why initialise it to empty string?

File: src/main/common/cookieProperties.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default class CookieProperties {
+
+  static getCookieParameters () {
+    return {
```
Comment (timja-kainos) [inline src/main/common/cookieProperties.ts]: this doesn't look like secure has been set? or sameSite?

Comment (timja-kainos) [reply src/main/common/cookieProperties.ts]: we usually use the cookies library which sets cookies secure by default, (doesn't do sameSite though)
https://github.com/pillarjs/cookies

File: src/main/app/forms/models/representativeDetails.ts (hunk 1 above)
Comment (robertparkinson) [reply src/main/app/forms/models/representativeDetails.ts]: it was having issues with accessing undefined. I'll double check to see if it is still needed

File: src/main/common/cookieProperties.ts (hunk 2 above)
Comment (robertparkinson) [reply src/main/common/cookieProperties.ts]: have put in sameSite. cookieParser uses Cookie underneath which has a sameSite option:
https://github.com/jshttp/cookie

**Pull request merged**